### PR TITLE
Rayleigh damping and HHT time integration added to Tensor mechanics m…

### PR DESCRIPTION
**Pull request by sveerara**
…odules StressDivergenceTensors.C and SolidMechInertialForceTensors.C #5559

### CODE REVIEW

File: modules/tensor_mechanics/include/kernels/StressDivergenceTensors.h [hunk 1]
```diff
@@ -28,6 +28,7 @@ class StressDivergenceTensors : public Kernel
 {
 public:
   StressDivergenceTensors(const InputParameters & parameters);
+  StressDivergenceTensors(const std::string & deprecated_name, InputParameters parameters); // DEPRECATED CONSTRUCTOR
```
Comment (dschwen) [inline modules/tensor_mechanics/include/kernels/StressDivergenceTensors.h]: You are reintroducing a deprectated constructor that was just removed.

File: modules/tensor_mechanics/src/kernels/StressDivergenceTensors.C [hunk 2]
```diff
@@ -29,6 +31,7 @@ InputParameters validParams<StressDivergenceTensors>()
 StressDivergenceTensors::StressDivergenceTensors(const InputParameters & parameters) :
     Kernel(parameters),
     _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _stress_old(getMaterialPropertyOldByName<RankTwoTensor>(_base_name + "stress")),
```
Comment (dschwen) [inline modules/tensor_mechanics/src/kernels/StressDivergenceTensors.C]: No. Please do not just add stateful materials to our main tensor mechanics kernel! These come at a cost! Please conditionally request staeful properties if they are _really_ needed.

File: modules/tensor_mechanics/src/kernels/StressDivergenceTensors.C [hunk 3]
```diff
@@ -79,13 +84,19 @@ StressDivergenceTensors::StressDivergenceTensors(const InputParameters & paramet
 Real
 StressDivergenceTensors::computeQpResidual()
 {
-  return _stress[_qp].row(_component) * _grad_test[_i][_qp];
+  if ((_dt > 0) && ((_alpha != 0) || (_zeta != 0)))
```
Comment (dschwen) [inline modules/tensor_mechanics/src/kernels/StressDivergenceTensors.C]: Stuff like this needs to have comments! Also the alpha!=0 || zeta!= 0 can be folded into a bool flag with a _meaningful_ name at construction time.
Now that I think about this, don't do any of this in ```StressDivergenceTensors```! Create a derived class. We do not want our classes to grow to become monolithic multipurpose monsters like in certain other mechanics systems ;-).
This is a straight forward case wher yuor derived class can add a few variables and overwrite the computeQpResidual et al. methods.

File: modules/tensor_mechanics/src/materials/ComputeLinearElasticStress.C [hunk 4]
```diff
@@ -16,6 +16,7 @@ InputParameters validParams<ComputeLinearElasticStress>()
 
 ComputeLinearElasticStress::ComputeLinearElasticStress(const InputParameters & parameters) :
     ComputeStressBase(parameters),
+    _stress_old(declarePropertyOld<RankTwoTensor>(_base_name + "stress")),
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/ComputeLinearElasticStress.C]: Same thing here. Do this in a class that is derived from ComputeLinearElasticStress. Adding stateful materials adds a **substantial** memory overhead to a material! And we do not need them for many applications.

File: modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+InputParameters validParams<NewmarkAccelAux>();
+
+/**
+ * Accumulate values from one auxiliary variable into another
+ */
+class NewmarkAccelAux : public AuxKernel
+{
+public:
+
+  /**
```
Comment (dschwen) [inline modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h]: We do not need to keep these boiler-plate comments around.

File: modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
```
Comment (dschwen) [inline modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h]: Remove empty lines please

File: modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+template<>
+InputParameters validParams<NewmarkAccelAux>();
+
+/**
+ * Accumulate values from one auxiliary variable into another
+ */
+class NewmarkAccelAux : public AuxKernel
+{
+public:
+
```
Comment (dschwen) [inline modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h]: Remove empty lines after access qualifiers, please.

File: modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  virtual ~NewmarkAccelAux() {}
+
+protected:
+  virtual Real computeValue();
+
+  VariableValue & _disp_old;
+  VariableValue & _disp;
+  VariableValue & _vel_old;
+  Real _beta;
+
```
Comment (dschwen) [inline modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h]: Remove empty lines before curly closing brackets, please

File: modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+#endif //NewmarkAccelAux_H
```
Comment (dschwen) [inline modules/tensor_mechanics/include/auxkernels/NewmarkAccelAux.h]: That symbol should be all caps, even in the comment

File: modules/tensor_mechanics/include/auxkernels/NewmarkVelAux.h [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+
+#ifndef NEWMARKVELAUX_H
+#define NEWMARKVELAUX_H
+
+#include "AuxKernel.h"
+
+
```
Comment (dschwen) [inline modules/tensor_mechanics/include/auxkernels/NewmarkVelAux.h]: No double empty lines, please. (and the other empty line comments from above apply as well)

File: modules/tensor_mechanics/include/kernels/SolidMechInertialForceTensors.h [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  const MaterialProperty<Real> & _density;
+  const VariableValue & _u_old;
+  const VariableValue & _vel_old;
+  const VariableValue & _accel_old;
+  const Real _beta;
+  const Real _gamma;
+  const Real _eta;
+
+};
+#endif //SOLIDMECHINERTIALFORCETENSORS_H
```
Comment (dschwen) [inline modules/tensor_mechanics/include/kernels/SolidMechInertialForceTensors.h]: empty line before this for consistency

File: modules/tensor_mechanics/src/auxkernels/NewmarkVelAux.C [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+
+#include "NewmarkVelAux.h"
+
+template<>
+InputParameters validParams<NewmarkVelAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+    params.addRequiredCoupledVar("acceleration","acceleration variable");
+    //params.addRequiredParam<Real>("beta","beta parameter");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/auxkernels/NewmarkVelAux.C]: Please remove commented out code

File: modules/tensor_mechanics/src/auxkernels/NewmarkVelAux.C [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+{
+}
+
+Real
+NewmarkVelAux::computeValue()
+{
+  Real vel_old = _u_old[_qp];
+  if (!isNodal())
+    mooseError("must run on a nodal variable");
+  return vel_old + (_dt*(1-_gamma))*_accel_old[_qp] + _gamma*_dt*_accel[_qp];
```
Comment (dschwen) [inline modules/tensor_mechanics/src/auxkernels/NewmarkVelAux.C]: For new code we try to keep whitespace around binary operators (at least in the phase_field module)

File: modules/tensor_mechanics/src/kernels/SolidMechInertialForceTensors.C [hunk 14]
```diff
@@ -0,0 +1,53 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "SolidMechInertialForceTensors.h"
```
Comment (dschwen) [inline modules/tensor_mechanics/src/kernels/SolidMechInertialForceTensors.C]: Can't we just call this Kernel ```InertialForce```? Having an abbreviated _solid mechanics_ and a _tensors_ (even though it neither is part of the solid mechanics module, nor does the kernel compute any tensors) seems a little confusing.

Comment (sveerara) [inline modules/tensor_mechanics/src/kernels/SolidMechInertialForceTensors.C]: Thanks for the comments. I will edit the files and resubmit.

**Comment by dschwen** [comment]
@sveerara is this ```NewmarkAccelAux``` kernel different from the one in solid machanics? Either way, you'll need to rename this because we have a bunch of apps that include both tensor_mechanics and solid_mechanics (the combined modules test app for example).

**Comment by jwpeterson** [comment]
I'm closing this PR because it looks like it was abandoned and/or the work may have already been merged in another branch.

**Pull request closed without merging**